scripts/HC01-JP/c100287004.lua

```diff
@@ -0,0 +1,127 @@
+	e3:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+	e3:SetCode(EFFECT_CANNOT_ACTIVATE)
+	e3:SetRange(LOCATION_MZONE)
+	e3:SetTargetRange(0,1)
+	e3:SetCondition(s.actcon)
+	e3:SetValue(1)
+	c:RegisterEffect(e3)
+	--Return to Hand
+	local e4=Effect.CreateEffect(c)
+	e4:SetDescription(aux.Stringid(id,1))
+	e4:SetCategory(CATEGORY_TOHAND)
+	e4:SetType(EFFECT_TYPE_SINGLE+EFFECT_TYPE_TRIGGER_O)
+	e4:SetCode(EVENT_BATTLE_DESTROYING)
+	e4:SetCondition(s.rhcon)
+	e4:SetTarget(s.rhtg)
+	e4:SetOperation(s.rhop)
+	c:RegisterEffect(e4)
+end
+function s.thcon(e,tp,eg,ep,ev,re,r,rp)
+	return Duel.GetTurnPlayer()==tp and Duel.GetFieldGroupCount(tp,LOCATION_DECK,0)>0
+		and Duel.GetDrawCount(tp)>0
+end
+function s.thcost(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return not e:GetHandler():IsPublic() end
+end
+function s.thfilter(c)
+	return bit.band(c:GetType(),0x82)==0x82 and c:IsAbleToHand()
+end
+function s.thtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return Duel.IsExistingMatchingCard(s.thfilter,tp,LOCATION_DECK,0,1,nil) end
+	local dt=Duel.GetDrawCount(tp)
+	if dt~=0 then
+		aux.DrawReplaceCount=0
+		aux.DrawReplaceMax=dt
+		local e1=Effect.CreateEffect(e:GetHandler())
+		e1:SetType(EFFECT_TYPE_FIELD)
+		e1:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+		e1:SetCode(EFFECT_DRAW_COUNT)
+		e1:SetTargetRange(1,0)
+		e1:SetReset(RESET_PHASE+PHASE_DRAW)
+		e1:SetValue(0)
+		Duel.RegisterEffect(e1,tp)
+		local cid=Duel.GetChainInfo(0,CHAININFO_CHAIN_ID)
+		local e2=Effect.CreateEffect(e:GetHandler())
+		e2:SetType(EFFECT_TYPE_FIELD+EFFECT_TYPE_CONTINUOUS)
+		e2:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+		e2:SetCode(EVENT_CHAIN_SOLVING)
+		e2:SetLabel(cid)
+		e2:SetLabelObject(e1)
+		e2:SetReset(RESET_PHASE+PHASE_DRAW)
+		e2:SetCondition(s.checkcon)
+		e2:SetOperation(s.checkop)
+		Duel.RegisterEffect(e2,tp)
+	end
+	Duel.SetOperationInfo(0,CATEGORY_TOHAND,nil,1,tp,LOCATION_DECK)
+end
+function s.checkcon(e,tp,eg,ep,ev,re,r,rp)
+	local cid,orig_effect=Duel.GetChainInfo(ev,CHAININFO_CHAIN_ID,CHAININFO_TRIGGERING_EFFECT)
+	return cid==e:GetLabel() and not e:GetOwner():IsRelateToEffect(orig_effect)
+end
+function s.checkop(e,tp,eg,ep,ev,re,r,rp)
+	e:GetLabelObject():Reset()
+	e:Reset()
+end
+function s.thop(e,tp,eg,ep,ev,re,r,rp)
+	aux.DrawReplaceCount=aux.DrawReplaceCount+1
+	if aux.DrawReplaceCount>aux.DrawReplaceMax or not e:GetHandler():IsRelateToEffect(e) then return end
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_ATOHAND)
+	local g=Duel.SelectMatchingCard(tp,s.thfilter,tp,LOCATION_DECK,0,1,1,nil)
+	if g:GetCount()>0 then
+		Duel.HintSelection(g)
+		Duel.SendtoHand(g,nil,REASON_EFFECT)
+	end
+end
+function s.actcon(e)
+	local ph=Duel.GetCurrentPhase()
+	return Duel.GetTurnPlayer()==e:GetHandler():GetControler() and ph>=PHASE_BATTLE_START and ph<=PHASE_BATTLE
+end
+function s.valcheck(e,c)
+	local mg=c:GetMaterial()
+	if mg:IsExists(Card.IsType,1,nil,TYPE_NORMAL) then
+		e:GetLabelObject():SetLabel(1)
+	else
+		e:GetLabelObject():SetLabel(0)
+	end
+end

```

Comment:
Where is the call to this function?
```suggestion
function s.valcheck(e,c)
	local mg=c:GetMaterial()
	if mg:IsExists(Card.IsType,1,nil,TYPE_NORMAL) then
		e:GetHandler():RegisterFlagEffect(id,RESET_EVENT+RESETS_STANDARD,0,1)
	end
end

```